cleanlab/multilabel_classification.py

```diff
@@ -0,0 +1,120 @@
+# Copyright (C) 2017-2022  Cleanlab Inc.
+# This file is part of cleanlab.
+#
+# cleanlab is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# cleanlab is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with cleanlab.  If not, see <https://www.gnu.org/licenses/>.
+
+"""
+Methods to rank the severity of label issues in multi-label classification datasets.
+Here each example can belong to one or more classes, or none of the classes at all.
+Unlike in multi-class classification, predicted class probabilities from model need not sum to 1 for each row.
+"""
+
+import numpy as np
+from typing import List
+
+from cleanlab.internal.validation import assert_valid_inputs
+from cleanlab.internal.util import get_num_classes
+from cleanlab.internal.multilabel_scorer import MultilabelScorer, ClassLabelScorer, Aggregator
+from cleanlab.internal.multilabel_utils import int2onehot
+
+
+def get_label_quality_scores(
+    labels: List,
+    pred_probs: np.ndarray,
+    *,
+    method: str = "self_confidence",
+    adjust_pred_probs: bool = False,
+    aggregator_kwargs: dict = {"method": "exponential_moving_average", "alpha": 0.8}
+) -> np.ndarray:
+    """Returns a label quality score for each datapoint.
+
+    This is a function to compute label quality scores for multi-label classification datasets,
+    where lower scores indicate labels more likely to contain an error.
+
+    Score is between 0 and 1 with lower values indicating labels inferred to be worse.
+    For each example, this method internally computes a separate score for each individual class
+    and then aggregates these per-class scores into an overall label quality score for the example.
+
+    Parameters
+    ----------
+    labels : List[List[int]]
```

Comment:
Shouldn't we let the autodoc-typehints handle this?

Reply:
I actually don't think we want to declare `labels: List[List[int]]` in the signature because technically `List[np.ndarray]` would be ok too. Hence I advocate leaving this as it is currently in my PR